src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/TagletManager.java

```diff
@@ -311,16 +311,14 @@ public void addNewSimpleCustomTag(String tagName, String header, String location
         if (tagName == null || locations == null) {
             return;
         }
-        Taglet tag = allTaglets.get(tagName);
+        Taglet tag = allTaglets.remove(tagName);
```

Comment:
The logic was somewhat clearer before. This code may be correct, but it would help to have an explanatory comment.
Note this code is only executed once per run of javadoc, so performance is definitely not an issue.

Reply:
Hm.  For me the previous code is confusing. Subsequent remove+put by the same key looks like some leftovers after refactoring.
Javadoc of this method is very explanatory. New code just repeats what is written there.  Not sure if additional comment required.

Reply:
In a context of a linked hash map, remove+put has the effect of reordering. In your refactor there's still a remove+put on the same key, you essentially just hoisted the two common occurrences from both branches to outside the if statement.

Not to _necessarily_ appeal to authority, but Jon is a true multi-decade veteran of Java language tooling and knows more about Javadoc than most of us combined, so if all he asks for is a comment, I'd add a comment :-). Maybe something to the effect of

```
// remove + put in both branches below move the tag to the back of the map's ordering
```

Reply:
comment added

Reply:
@jonathan-gibbons are you good with current code?

Reply:
I reread the code in a more complete context, and the more detailed description in the doc comment for the method. The proposed comment isn't great, since it hints at the existence of a tag when there might not be one (i.e. `tag == null`).

The `then` part is sufficiently obvious, since the code is inserting a new `SimpleTaglet`, so ,I'd suggest to just enhance the `else` comment to:

```
// Move existing tag to the back
```